### PR TITLE
feat: Add indexer.toml file

### PR DESCRIPTION
**Pull request by roninjin10**
- adds toml file for configuring indexer

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5828** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5828" target="_blank">Graphite</a> 
    * **PR #5829** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5829" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5829?utm_source=stack-comment).

File: indexer/config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	// Load the config.
+	conf, err := config.LoadConfig(tmpfile.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the configuration was loaded correctly.
+	if conf.Chain.Preset != 1234 {
+		t.Errorf("Expected chain preset 1234, got %d", conf.Chain.Preset)
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply indexer/config/config_test.go:57]: Nice tip!

[the remark this answers is not in the document]
Comment (roninjin10) [reply indexer/config/config_test.go]: done

File: indexer/config/config.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package config
+
+import (
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+// The go struct representation of the `indexer.toml` file used to configure the indexer
```
Comment (tynes) [inline indexer/config/config.go:9]: godoc usually starts with the name of the struct/function
ie
```
Config represents ...
```

File: indexer/cmd/indexer-refresh/main.go [hunk 3]
```diff
@@ -37,7 +47,7 @@ func main() {
 		"by account on L1 and L2"
 
 	app.Action = indexer.Main(GitVersion)
-	err := app.Run(os.Args)
+	err = app.Run(os.Args)
```
Comment (tynes) [inline indexer/cmd/indexer-refresh/main.go:50]: could do something like:

```golang
if err := app.Run(os.Args); err != nil {
  log.Crit("...")
}
```

to keep the second error only in scope of the branch

File: indexer/cmd/indexer-refresh/main.go [hunk 4]
```diff
@@ -28,6 +29,15 @@ func main() {
 		),
 	)
 
+	// TODO don't hardcode this
```
Comment (roninjin10) [inline indexer/cmd/indexer-refresh/main.go:32]:
```suggestion
	// TODO https://linear.app/optimism/issue/DX-55/api-implement-rest-api-with-mocked-data
	// don't hardcode this
```

File: indexer/indexer.toml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+[chain]
+preset = 420
+
+[rpcs]
+l1-rpc = "${INDEXER_RPC_URL_L1}"
+l2-rpc = "${INDEXER_RPC_URL_L2}"
+
```
Comment (roninjin10) [inline indexer/indexer.toml:7]: rpc.l1-rpc is redundant.    can just be l1-uri or just l1

**Review by hamdiallam — APPROVED** [review]
sweet!

**Pull request merged**